### DISCUSSION TITLE
Need some help with lua/mympd_state

### DISCUSSION THREAD

**Discussion by chiefy** [jcorporation/myMPD#1350] [Q&A] [answered]
Using the latest release. This was working and I am not sure if I updated my docker image and things stopped working - or something else. Seems really bizarre, but my listenbrainz Lua script stopped working with the following error:

```
 attempt to concatenate a nil value (field 'listenbrainz_token')
```
The relevant line in my lua:

```
mympd.init()

uri = "https://api.listenbrainz.org/1/submit-listens"
headers = "Content-type: application/json\r\n"..
  "Authorization: Token "..mympd_state["listenbrainz_token"].."\r\n"
```

In my state dir I have a file named `listenbrainz_token` with my api token. 

As I said this was working until a day or two ago so I am not sure what changed. I ended up dumping out my `mympd_state` to console and get this:

```
state:	{ ["mympd_uri_plain"] = http://172.23.0.4:80/,["partition"] = default,["jukebox_ignore_hated"] = false,["total_time"] = 205,["mympd_uri"] = http://172.23.0.4:80/,["jukebox_max_song_duration"] = 0,["jukebox_min_song_duration"] = 0,["repeat"] = false,["jukebox_uniq_tag"] = Artist,["jukebox_playlist"] = Database,["mixrampdelay"] = -1537353130755225596,["jukebox_last_played"] = 24,["queue_version"] = 2,["consume_state"] = 0,["next_song_id"] = -1,["next_song_pos"] = -1,["jukebox_mode"] = 0,["volume"] = 67,["auto_play"] = false,["play_state"] = 2,["crossfade"] = 0,["single_state"] = 0,["music_directory"] = ,["replaygain"] = 0,["song_id"] = 11,["mixrampdb"] = 6413300356951836676,["playlist_directory"] = ,["elapsed_time"] = 102,["song_pos"] = 10,["random"] = false,["jukebox_queue_length"] = 1,["queue_length"] = 11,} 
```

I was digging through the code a little but C isn't my thing really - not sure where custom state vars get initialized etc. A little pointer/help would be great, thanks!

**Comment by jcorporation** [comment, marked as answer]
The config Setting for the listenbrainz Token was removed. The state file is obsolete. 

You should create a variable in the gui. This variables are accessible through the mympd_env lua table as described here: https://jcorporation.github.io/myMPD/scripting/

**Reply by chiefy** [reply]
Ah! ok thanks @jcorporation. I guess I must've had a old release and I think I do remember updating my docker image at some point but never deploying it. Must've just been that. Anyhow, i switched to use the variable and it's working great again. Thank you!
